### PR TITLE
[CI-7494] Add mount cert command and break into tabs

### PR DESCRIPTION
**Pull request by aimurphy**
Preview site menu path: Documentation > Continuous Integration > Use CI > Set Up Build Infrastructure > Set up a local runner build infrastructure

* Added step for self-signed certs command
* Created three tabs containing the separate Delegate and Drone Runner instructions for Linux, macOS, and Windows.

Menu cascade for preview site: Documentation > Continuous Integration > Use CI > Set up Build Infrastructure > Set up a local runner build infrastructure
https://642324e7534eae00a330676b--harness-developer.netlify.app/docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure

### CODE REVIEW

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 1]
```diff
@@ -44,29 +50,174 @@ Make sure to create the delegate at the appropriate scope, such as the project l
...
+2. To use self-signed certificates, export `CI_MOUNT_VOLUMES` along with a comma-separated list of source paths and destination paths formatted as `source_path:destination_path`, for example:
+
+   ```
+   export CI_MOUNT_VOLUMES=[path_to_cert_on_local]:/etc/ssl/certs/ca-certificats.crt, [source_path_2]:[destination_path_2]
+   ```
+
+3. Run the following command to enable execution permissions for the Runner:
+
+   ```
+   chmod +x drone-docker-runner-windows-amd64
```
Comment (devkimittal) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:187]: not needed for windows

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 2]
```diff
@@ -44,29 +50,174 @@ Make sure to create the delegate at the appropriate scope, such as the project l
...
+You must run the Drone Runner executable from a separate machine than the one that your Delegate is running on. Make sure to run these commands on the appropriate machine.
+
+:::
+
+1. Download a [Drone Runner executable](https://github.com/harness/drone-docker-runner/releases).
+2. Open a terminal with Administrator privileges.
+2. To use self-signed certificates, export `CI_MOUNT_VOLUMES` along with a comma-separated list of source paths and destination paths formatted as `source_path:destination_path`, for example:
+
+   ```
+   export CI_MOUNT_VOLUMES=[path_to_cert_on_local]:/etc/ssl/certs/ca-certificats.crt, [source_path_2]:[destination_path_2]
```
Comment (devkimittal) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:181]: replace export by SET

**Comment by rajatharanganath** [comment]
@devkimittal - Main concern from customer is on Windows set-up. We need to articulate that delegate needs to be installed on a linux machine, as delegate is not compatible for windows at this point. Can you please review windows set-up and suggest more details on it?

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md (hunk 2 above)
Comment (aimurphy) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:181]: do I change export to set for all 3 or only windows?

**Comment by aimurphy** [comment]
> @devkimittal - Main concern from customer is on Windows set-up. We need to articulate that delegate needs to be installed on a linux machine, as delegate is not compatible for windows at this point. Can you please review windows set-up and suggest more details on it?

Which DELEGATE_TAG should you use? One of the Linux ones?

Proposed revisions for windows instructions:

~~~
## Prepare machines

To configure a local runner build infrastructure for Windows, you need two machines:

* A Windows machine where the Drone Runner will run.
* A Linux machine where the Delegate will run.

## Install the Delegate

On the Linux machine where you want to run the Delegate, use the following modifications along with the **Docker environment** instructions in [Install a Delegate](/docs/platform/Delegates/install-delegates/install-a-delegate):

* Add `-e DELEGATE_TAGS="<delegate-tag>"`. Use one of the following tags: `linux-amd64` or `linux-arm64`.
* Add `-e RUNNER_URL=http://[windows_machine_hostname_or_ip]:3000`.

:::caution

The `RUNNER_URL` must point to the Windows machine where the Drone Runner will run.

:::

Here's an example of an install script for a Linux amd64 machine that points to the Windows machine where the Drone Runner will run:

```
docker run --cpus=1 --memory=2g \
  -e DELEGATE_NAME=docker-delegate \
  -e NEXT_GEN="true" \
  -e DELEGATE_TYPE="DOCKER" \
  -e ACCOUNT_ID=H5W8iol5TNWc4G9h5A2MXg \
  -e DELEGATE_TOKEN=ZWYzMjFmMzNlN2YxMTExNzNmNjk0NDAxOTBhZTUyYzU= \
  -e DELEGATE_TAGS="linux-amd64" \
  -e RUNNER_URL=http://[windows_machine_hostname_or_ip]:3000 \
  -e MANAGER_HOST_AND_PORT=https://app.harness.io harness/delegate:23.02.78306
```

Make sure to create the delegate at the appropriate scope, such as the project level or account level.

## Install the Drone Runner

The Drone Runner service performs the build work. The Delegate needs the Runner to run CI builds.

:::caution

Run the Drone Runner executable on the Windows machine that you specified in the Delegate's `RUNNER_URL`.

:::

1. On the target Windows machine where you want to run the Drone Runner, download a [Drone Runner executable](https://github.com/harness/drone-docker-runner/releases).
2. Open a terminal with Administrator privileges.
3. To use self-signed certificates, set `CI_MOUNT_VOLUMES` along with a comma-separated list of source paths and destination paths formatted as `path/to/source:path/to/destination`, for example:

   ```
   set CI_MOUNT_VOLUMES=[path_to_cert_on_local]:/etc/ssl/certs/ca-certificats.crt, [path/to/source/2]:[path/to/destination/2]
   ```

4. Run the following command to start the runner binary:

   ```
   drone-docker-runner-windows-amd64.exe server
   ```

Here is an example of the commands to install the Drone Runner with self-signed certificates:

```
set CI_MOUNT_VOLUMES=<path-to-cert-on-local>:/etc/ssl/certs/ca-certificates.crt
./drone-docker-runner-windows-amd64 server
```

~~~

**Comment by aimurphy** [comment]
@devkimittal , @shubham149  I need your feedback on: https://github.com/harness/developer-hub/pull/1100#issuecomment-1483202838

**Comment by vistaarjuneja** [comment]
Hey @aimurphy , the delegate tag is related with what we specify in the pipeline UI. It should be whatever infrastructure the builds are running on, since internally we might modify the run step to be compatible with the infrastructure.
So for windows, the delegate tag should be `windows-amd64` which should be the same as that provided in the pipeline UI.
One small nit: CI_MOUNT_VOLUMES should not have a space after the comma as that might not work.

**Review by douglas-j-bothwell — APPROVED** [review]
Approved with one sug.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 3]
```diff
@@ -8,23 +8,31 @@ helpdocs_is_private: false
...
 ## Install the Delegate
 
 Use the following modifications along with the **Docker environment** instructions in [Install a Delegate](/docs/platform/Delegates/install-delegates/install-a-delegate):
 
-* Add `-e DELEGATE_TAGS="<delegate-tag>"`. Use one of the following tags: `macos-amd64`, `macos-arm64`, `windows-amd64`, `linux-amd64`, `linux-arm64`.
-* For macOS, add `-e RUNNER_URL=http://host.docker.internal:3000`.
-* For Linux, add `--net=host` to the first line.
-* For Windows, add `-e RUNNER_URL=http://<windows-machine-hostname-or-ip>:3000`. For Windows, the Drone Runner must run on a separate machine than the one that your Delegate runs on. This variable must point to the Drone Runner's machine.
+* Add `--net=host` to the first line.
+* Add `-e DELEGATE_TAGS="<delegate-tag>"`. Use one of the following tags: `linux-amd64` or `linux-arm64`.
```
Comment (shubham149) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:33]: This does not mention when to specify linux-amd64 / linux-arm64

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 4]
```diff
@@ -37,36 +45,187 @@ docker run --cpus=1 --memory=2g --net=host \
...
+
+```mdx-code-block
+  </TabItem>
+  <TabItem value="macOS" label="macOS">
+```
+## Install the Delegate
+
+Use the following modifications along with the **Docker environment** instructions in [Install a Delegate](/docs/platform/Delegates/install-delegates/install-a-delegate):
+
+* Add `-e DELEGATE_TAGS="<delegate-tag>"`. Use one of the following tags: `macos-amd64` or `macos-arm64`.
```
Comment (shubham149) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:89]: Ditto

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 5]
```diff
@@ -37,36 +45,187 @@ docker run --cpus=1 --memory=2g --net=host \
...
+  -e MANAGER_HOST_AND_PORT=https://app.harness.io harness/delegate:23.02.78306
+```
+
+Make sure to create the Delegate at the appropriate scope, such as the project level or account level.
+
+## Install the Drone Runner
+
+The Drone Runner service performs the build work. The Delegate needs the Runner to run CI builds.
+
+1. Download a [Drone Runner executable](https://github.com/harness/drone-docker-runner/releases).
```
Comment (shubham149) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:112]: This does not mention that customer need to download a version with correct os and arch

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 6]
```diff
@@ -37,36 +45,187 @@ docker run --cpus=1 --memory=2g --net=host \
...
+2. To use self-signed certificates, export `CI_MOUNT_VOLUMES` along with a comma-separated list of source paths and destination paths formatted as `path/to/source:path/to/destination`, for example:
```
Comment (shubham149) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:113]: Maybe provide a link for self signed cert page with more details on when to use it

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md [hunk 7]
```diff
@@ -37,36 +45,187 @@ docker run --cpus=1 --memory=2g --net=host \
...
+
+   ```
+   export CI_MOUNT_VOLUMES="[path/to/local/cert]:/etc/ssl/certs/ca-certificates.crt,[path/to/local/cert2]:/etc/ssl/certs/cacerts.pem"
+   ```
+
+3. Enable execution permissions for the Runner. For example:
+
+   ```
+   sudo chmod +x drone-docker-runner-linux-arm64
```
Comment (shubham149) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:64]: example for docker delegate is of linux amd64 but runner is of arm

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md (hunk 6 above)
Comment (aimurphy) [reply docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure.md:113]: Which page?

**Pull request merged**